test/integration/publishing_event_processor_test.rb

```diff
@@ -0,0 +1,64 @@
+require 'integration_test_helper'
+require 'bunny-mock'
+require 'govuk_index/publishing_event_processor'
+require 'govuk_message_queue_consumer'
+
+class GovukIndex::PublishingEventProcessorTest < IntegrationTest
+  def setup
+    bunny_mock = BunnyMock.new
+    channel = bunny_mock.start.channel
+
+    consumer = GovukMessageQueueConsumer::Consumer.new(
+      queue_name: "bigwig.test",
+      processor: GovukIndex::PublishingEventProcessor.new("govuk_test"),
+      rabbitmq_connection: bunny_mock
+    )
+
+    @queue = channel.queue("bigwig.test")
+    consumer.run
+  end
+
+  def test_should_save_new_document_to_elasticsearch
+    fixture_file = File.expand_path("../fixtures/publishing_api_payload.json", __FILE__)
+    payload = JSON.parse(File.read(fixture_file))
+    @queue.publish(payload)
+
+    document = fetch_document_from_rummager(id: "/aaib-reports/pitts-s-2b-g-skyd-21-june-1996", index: "govuk_test")
+
+    assert_equal "/aaib-reports/pitts-s-2b-g-skyd-21-june-1996", document["_source"]["link"]
+    assert_equal "/aaib-reports/pitts-s-2b-g-skyd-21-june-1996", document["_id"]
+    assert_equal "aaib_report", document["_type"]
+  end
+
+  def test_should_raise_error_when_invalid_payload
```

Comment:
I would prefer to see the validation checking in the unit tests. I don't feel this is necessarily worth an integration test.